text/0055-signature-based-finality.md

```diff
@@ -32,43 +32,42 @@ The current process of block finalization is this:
 5. Consensus creates a finality signature for the block.
 6. Linear chain stores the signature and gossips it.
 
-In order to only consider fully signed blocks as finalized, we'll have to change this process to the following:
+We wouldn't change the meaning of "finalized" in order not to introduce confusion - a block being finalized would still mean finalized by consensus. However, we would add a new, "signed" status of the block, which would be used instead of "finalized" for some purposes (for example, advancing to the next era will require the switch block to be signed, instead of only finalized). The process would then look the following way:
 
 1. A `BlockPayload` gets finalized by a consensus instance.
 2. The consensus component announces a `FinalizedBlock`.
 3. Contract runtime executes the block and announces a `Block`.
-4. In response to the announcement, linear chain component caches the block, and consensus creates a finality signature.
-5. Linear chain stores the signature and gossips it. It also receives signatures sent over the network.
-6. Once the linear chain collects enough signatures for the block, it considers it finalized.
+4. In response to the announcement, linear chain component caches the block and announces that it was added.
+5. Consensus creates a finality signature for the block.
+6. Linear chain stores the signature and gossips it. It also receives signatures sent over the network.
+7. Once the linear chain collects enough signatures for the block, the block is considered signed.
     - If the block is unknown yet, it requests the block from the sender of the last signature.
```

Comment:
And after a timeout, I guess we'd request it from the sender of the next signature and so on? It should probably work in a similar way to how the block validator downloads deploys.